build.php

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env php
+<?php
+
+use Secondtruth\Compiler\Compiler;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+
+if(!defined('PACKAGE_ROOT')) define('PACKAGE_ROOT', dirname(__FILE__) . '/');
+if(!defined('BUILD_FOLDER')) define('BUILD_FOLDER', tempnam(sys_get_temp_dir(),'') );
+
+require_once(PACKAGE_ROOT . '/vendor/autoload.php');
+
+$fs = new Filesystem();
+
+// Make sure our build folder exists
+if ($fs->exists(BUILD_FOLDER)) { $fs->remove(BUILD_FOLDER); }
+
+$fs->mkdir(BUILD_FOLDER);
+
+// Set up our build folder
+$fs->mirror(PACKAGE_ROOT . 'bin', BUILD_FOLDER . '/bin');
+$fs->mirror(PACKAGE_ROOT . 'src', BUILD_FOLDER . '/src');
+$fs->copy(PACKAGE_ROOT . 'composer.json', BUILD_FOLDER . '/composer.json');
+$fs->copy(PACKAGE_ROOT . 'composer.lock', BUILD_FOLDER . '/composer.lock');
+
+// Install dependencies
+$composerStatement = sprintf(
+    'composer install --prefer-dist --no-dev --optimize-autoloader --optimize-autoloader --working-dir=%s',
+    BUILD_FOLDER
+);
+$process = new Process($composerStatement);
+$process->run();
+
+// Compile the executable
+$compiler = new Compiler(BUILD_FOLDER);
+$compiler->addIndexFile('bin/upgrade-code');
+$compiler->addDirectory('src');
+$compiler->addDirectory('vendor');
+
+$compiledPath = PACKAGE_ROOT . "upgrade-code.phar";
+if ($fs->exists($compiledPath)) { $fs->remove($compiledPath);}
+@$compiler->compile($compiledPath);
```

Comment:
we silence errors here but then proceed as if nothing went wrong (if it did) is that intentional / the best way to do this?

Reply:
There's some silly warning getting thrown here. Looks like the problem is with the third party library I'm using. I'll try doing a PR against that third party library.

Reply:
The problem is not with the third party library per say. It calls `token_get_all` on each file which throws warning on some minor syntax errors in a few third party libs. I've suppress those specific warning instead.